### DISCUSSION TITLE
basic help with ai and refining image

### DISCUSSION THREAD

**Discussion by itchymitchy** [Acly/krita-ai-diffusion#824] [Q&A]
I am generating some bar background images that I can use in comps, using the prompt 'bar background'. Everytime I generate, there is a drink sitting on the bar. So I select an area around the drink and I'm trying to get the ai to remove it and replace with empty background consistent with the rest of the image. I just can't get this to work, either I get a blank white patch, or it creates a different drink. I've tried all the different options like fill, add content, remove content, refine etc. I'm obviously missing something really basic. Can anyone help?

**Comment by Hypersonic-Popcorn** [comment]
I would draw a crude representation of what I wanted in the area. Like a color close to the bar top and wall in the area you'd like them in. Then instead of prompting for a bar, prompt for a wooden countertop and wall.

There are models that do exactly what you want, LAMA and MAT come to mind, but I don't know how easy it would be to bring them to this plugin. See [IOPaint](https://www.iopaint.com/) for examples. You can probably run IOPaint in the background if you don't mind doing so.

**Reply by itchymitchy** [reply]
Thanks so much for responding. I was missing a trick - I hadn't clicked apply on the original ai generated image. Once I did that I could then select round the drink on the bar, choose remove content without a prompt and the ai did what I wanted. Not quite as good as I could do in 5 mins in Photoshop, but hey at least it's working!

**Reply by minsky91** [reply]
Often the fastest way to remove an object (if it's small enough) from an image is to use Krita's clone brush and then Refine the patch with the plugin, if necessary. For midsize and big objects, Remove content remains the best option. It might take though quite a lot of generations and prompt adjustments, as well as trying various checkpoints, not to mention refining selection around the object, to have removed it cleanly. Occasionally, this operation will insert a new, surprising content in place of the unwanted object, sometimes something breathtakingly beautiful (when the spot in question is, say, a piece of a landscape or a room interior), so you want to keep it, as it enhances the image substantially. I now regularly use Remove content to add something to the picture that I wouldn't myself come up with. It seems to use a larger context area around the object than an average Refine operation, so the results are typically more organic to the image.

**Reply by Acly** [reply]
> There are models that do exactly what you want, LAMA and MAT come to mind, but I don't know how easy it would be to bring them to this plugin.

"Remove Object" uses MAT, same as when you choose "Fill: Inpaint". It's used as a pre-fill for the area though, with diffusion afterwards.
